src/solvers/dgmulti/dg_parabolic.jl

```diff
@@ -222,55 +273,84 @@ function calc_viscous_penalty!(scalar_flux_face_values, u_face_values, t, bounda
                                mesh, equations::AbstractEquationsParabolic, dg::DGMulti,
                                parabolic_scheme, cache, cache_parabolic)
   # compute fluxes at interfaces
-  @unpack scalar_flux_face_values, inv_h = cache_parabolic
-  @unpack mapM, mapP = mesh.md
+  (; scalar_flux_face_values) = cache_parabolic
+  (; mapM, mapP) = mesh.md
   @threaded for face_node_index in each_face_node_global(mesh, dg)
     idM, idP = mapM[face_node_index], mapP[face_node_index]
     uM, uP = u_face_values[idM], u_face_values[idP]
-    inv_h_face = inv_h[face_node_index]
-    scalar_flux_face_values[idM] = scalar_flux_face_values[idM] + penalty(uP, uM, inv_h_face, equations, parabolic_scheme)
+    scalar_flux_face_values[idM] = scalar_flux_face_values[idM] + penalty(uP, uM, equations, parabolic_scheme)
   end
   return nothing
 end
 
-
-function calc_divergence!(du, u::StructArray, t, flux_viscous, mesh::DGMultiMesh,
-                          equations::AbstractEquationsParabolic,
-                          boundary_conditions, dg::DGMulti, parabolic_scheme, cache, cache_parabolic)
-
-  @unpack weak_differentiation_matrices = cache_parabolic
-
-  reset_du!(du, dg)
+function calc_divergence_volume_integral!(du, u, flux_viscous, mesh::DGMultiMesh,
+                                          equations::AbstractEquationsParabolic,
+                                          dg::DGMulti, cache, cache_parabolic)
+  (; weak_differentiation_matrices) = cache_parabolic
 
   # compute volume contributions to divergence
   @threaded for e in eachelement(mesh, dg)
     for i in eachdim(mesh), j in eachdim(mesh)
       dxidxhatj = mesh.md.rstxyzJ[i, j][1, e] # assumes mesh is affine
       apply_to_each_field(mul_by_accum!(weak_differentiation_matrices[j], dxidxhatj),
-                                view(du, :, e), view(flux_viscous[i], :, e))
+                          view(du, :, e), view(flux_viscous[i], :, e))
     end
   end
+end
+
+function calc_divergence_volume_integral!(du, u, flux_viscous, mesh::DGMultiMesh{NDIMS, <:NonAffine},
+                                          equations::AbstractEquationsParabolic,
+                                          dg::DGMulti, cache, cache_parabolic) where {NDIMS}
+  (; weak_differentiation_matrices, dxidxhatj, local_flux_viscous_threaded) = cache_parabolic
+
+  # compute volume contributions to divergence
+  @threaded for e in eachelement(mesh, dg)
+
+    local_viscous_flux = local_flux_viscous_threaded[Threads.threadid()][1]
+    for i in eachdim(mesh)
+      # rotate flux to reference coordinates
+      fill!(local_viscous_flux, zero(eltype(local_viscous_flux)))
+      for j in eachdim(mesh)
+        @. local_viscous_flux = local_viscous_flux + dxidxhatj[j, i][:, e] * flux_viscous[j][:, e]
```

Comment:
Allocations?

Reply:
It does! That is frustrating - what is the point of having broadcast 

Fixing that (and a similar issue in the divergence calculations).